### ISSUE TITLE
WISH: convert base script + routines

### ISSUE THREAD

**Issue by rerpha** [ISISComputingGroup/IBEX#6694]
Currently waiting on a script to be sent. 

As an instrument scientist I would like my base script to be converted to Python so I can use it in IBEX. WISH workflow is usually they give users a base script then modify depending on needs. 

### Acceptance criteria
- Script is converted and lives in instrument scripts repo
- convert the routines file

**Comment by rerpha** [comment]
reply from IS: 

> 1.	any script openGENIE in the U: drive would do, preferably one with a magnetic field and dilution fridge.

**Comment by rerpha** [comment]
Routines file: 

```gcl
#

#PE press pressure control scripts
PROCEDURE  set_pressure
PARAMETERS  setpoint=integer  rate=integer  mode=integer 
LOCAL minpress maxpress loopmode nompress apprate currentpress samecount lastpress maxloops
LOCAL vi_name="C:\\LabVIEW Modules\\Instruments\\PEARL\\PEARL Pressure Cell Controller\\PEARL Pressure Cell Controller - Front Panel.vi"

## Get the Pressure controller values
  minpress <~ GETLabVIEWvar(vi_name, "Minimum Pressure")
  nompress <~ GETLabVIEWvar(vi_name, "Nominal Pressure")
  maxpress <~ GETLabVIEWvar(vi_name, "Maximum Pressure")
  apprate  <~ GETLabVIEWvar(vi_name, "Application Rate")
  loopmode <~ GETLabVIEWvar(vi_name, "Servo Loop Mode")

printn "The current values are"
IF loopmode=1
printn minpress ", " nompress ", " maxpress ", " apprate ", closed loop mode "
ELSE 
printn minpress ", " nompress ", " maxpress ", " apprate ", open loop mode "
ENDIF

PushLabVIEWButton/wait  vi_name "Stop Button"


IF DEFINED(setpoint)

minpress=setpoint-2
nompress=setpoint
maxpress=setpoint+2
ENDIF

IF DEFINED(rate)
apprate=rate
ENDIF

 IF DEFINED(mode)
 loopmode=mode
 ENDIF

printn "The new values are"
IF loopmode=1
printn minpress ", " nompress ", " maxpress ", " apprate ", closed loop mode "
ELSE 
printn minpress ", " nompress ", " maxpress ", " apprate ", open loop mode "
ENDIF


SetLabVIEWvar vi_name "Minimum Pressure" minpress
SetLabVIEWvar vi_name "Nominal Pressure" nompress
SetLabVIEWvar vi_name "Maximum Pressure" maxpress
SetLabVIEWvar vi_name "Application Rate" apprate
SetLabVIEWvar vi_name "Servo Loop Mode"  loopmode

PushLabVIEWButton/wait vi_name "Send Parameters"
PushLabVIEWButton/wait vi_name "Start Button"

printn "Waiting for pressure to be reached"

#cset/control Pcell=nompress lowlimit=minpress highlimit=maxpress

maxloops=50
samecount=0
lastpress=0
LOOP

currentpress <~ GETLabVIEWvar(vi_name, "Current Pressure")
EXITIF (currentpress >= minpress) AND (currentpress <= maxpress) OR (samecount >= maxloops)

IF currentpress=lastpress
samecount=samecount+1
ELSE
samecount=0
ENDIF


sleep 10
printn "Pressure at " currentpress
lastpress=currentpress
ENDLOOP

printn "Reached pressure, now at " currentpress

printn ""
printn now()
printn ""

#cset/nocontrol Pcell=nompress 

ENDPROCEDURE

#
# Procedure to set pressure in closed loop mode
#
PROCEDURE pe_set_closed
PARAMETERS  setpoint=integer  rate=integer 
LOCAL mode

IF NOT DEFINED(setpoint); setpoint=INQUIRE("Pressure required="); ENDIF
IF NOT DEFINED(rate); rate=INQUIRE("Application rate="); ENDIF

mode=1
printn "Setting pressure in closed loop mode"

printn ""
printn now()
printn ""

set_pressure setpoint rate mode

ENDPROCEDURE
#End of PE press stuff
PROCEDURE waitformove
LOCAL moving i
	# Wait for move to start. If already in motion go to next stage
	moving=GetLabviewVar("c:\\LabVIEW Modules\\Drivers\\Galil DMC2280\\Galil - System Functions.llb\\Galil - Table of motor details.vi", "In Motion")
	IF moving=0
		LOOP i FROM 0 TO 100 
			moving=GetLabviewVar("c:\\LabVIEW Modules\\Drivers\\Galil DMC2280\\Galil - System Functions.llb\\Galil - Table of motor details.vi", "In Motion")	
			EXITIF moving=1
			SLEEP/MILLISEC 10
		ENDLOOP
	ENDIF
	
	# Wait for move to stop. If already stopped go to next stage
	moving=GetLabviewVar("c:\\LabVIEW Modules\\Drivers\\Galil DMC2280\\Galil - System Functions.llb\\Galil - Table of motor details.vi", "In Motion")
	IF moving=1
		LOOP i FROM 0 TO 50000 
			moving=GetLabviewVar("c:\\LabVIEW Modules\\Drivers\\Galil DMC2280\\Galil - System Functions.llb\\Galil - Table of motor details.vi", "In Motion")	
			EXITIF moving=0
			SLEEP/MILLISEC 10
		ENDLOOP
	ENDIF
ENDPROCEDURE
#
PROCEDURE START_DAE_MONITOR
LOCAL lv vi mon
mon="c:\\labview modules\\dae\\monitor\\dae_monitor.vi"
lv=_create_dcomlabview()
vi=dcom:call(lv,"GetViReference", "", mon)
IF (dcom:call:propget(vi, "ExecState") = 1)
    STARTVI mon
ENDIF
ENDPROCEDURE

PROCEDURE START_NGEM
LOCAL run
run = as_string(getrunnumber())
system("c:/Instrument/Apps/EPICS/base/master/bin/windows-x64/caput.exe TE:NDLT1177:NGEM_01:INSTRUN "+run)
system("c:/Instrument/Apps/EPICS/base/master/bin/windows-x64/caput.exe TE:NDLT1177:NGEM_01:START:SP 1")
ENDPROCEDURE

PROCEDURE STOP_NGEM
system("c:/Instrument/Apps/EPICS/base/master/bin/windows-x64/caput.exe TE:NDLT1177:NGEM_01:STOP:SP 1")
ENDPROCEDURE

#
START_DAE_MONITOR
#
cset/control ORC_cycle lowlimit=2 highlimit=4998
cset/control DecoupledCH4 lowlimit=45 highlimit=55
cset/control G2_TK_PIR highlimit=0.05
cset/control G7_GUIDE_TGT highlimit=0.5
cset/control G8_GUIDE_BCKH highlimit=0.1
#cset/control MOD_Runtime lowlimit=60 highlimit=1099
#cset/control MOD_uAh lowlimit=10 highlimit=679
cset/control ph_chop1_TGT lowlimit=1799 highlimit=1801
cset/control ph_chop2 lowlimit=4499 highlimit=4501
cset/control ph_chop3_TGT lowlimit=5884 highlimit=5886
cset/control ph_chop3_BCKHSE lowlimit=5904 highlimit=5906
cset/control ph_chp1_BCKHSE lowlimit=1819 highlimit=1821
```

**Comment by rerpha** [comment]
Taking my name off this as I am prioritising Beckhoff work for obvious reasons - should be fairly straightforward and only require the routines file to be converted and placed in the instrument scripts, and a simple script to be written for the scientists to tweak as needed.

**Comment by daryakoskeroglu** [comment]
PEARL Pressure Controller needs to be fully implemented to apply logic to converted routines. See the tickets here: #6763 and #6768

**Comment by daryakoskeroglu** [comment]
Created a new ticket to add VI buttons logic after their implementation in tickets above.
See ticket here: [#6897]

**Comment by daryakoskeroglu** [comment]
The branch routines pushed is : Ticket6694_convert_routines

**Comment by daryakoskeroglu** [comment]
`from genie_python import genie as g
from time import sleep
from datetime import datetime


def set_pressure(setpoint, rate, mode)
	
	minpress = g.get_block_value("Minimum Pressure")
	nompress = g.get_block_value("Nominal Pressure")
	maxpress = g.get_block_value("Maximum Pressure")
	apprate = g.get_block_value("Application Rate")
	loopmode = g.get_block_value("Servo Loop Mode")

	parameters = ['minpress', 'nompress', 'maxpress', 'apprate']
	print ("The current values are:", *parameters, sep="\n")

	if loopmode = 1:
		print ("closed loop mode")
	else 
		print ("open loop mode")
        
    #PushLabVIEWButton/wait  vi_name "Stop Button"

	if setpoint != None :
		minpress=setpoint-2
		nompress=setpoint
		maxpress=setpoint+2

	if rate != None:
		apprate = rate

	if mode != None:
		loopmode = mode
        
	print("The new values are:",*parameters, sep="\n")

	if loopmode = 1:
		print ("closed loop mode")
	else 
		print ("open loop mode")

	g.set_block_value("Minimum Pressure", minpress)
	g.set_block_value("Nominal Pressure", nompress)
	g.set_block_value("Maximum Pressure", maxpress)
	g.set_block_value("Application Rate", apprate)
	g.set_block_value("Servo Loop Mode", loopmode)

    #PushLabVIEWButton/wait vi_name "Send Parameters"
    #PushLabVIEWButton/wait vi_name "Start Button"
	print("Waiting for pressure to be reached")

	#g.cset(Pcell= nompress, lowlimit = minpress, highlimit = maxpress)

	maxloops=50
	samecount=0
	lastpress=0

	for i in range (0, maxloops+1):

		currentpress = g.get_block_value("Current Pressure")

		if(currentpress >= minpress) and (currentpress <= maxpress) or (samecount >= maxloops):
			exit
		if currentpress = lastpress:
			samecount += 1
		else:
			samecount = 0
		sleep(10)
		print (f"Pressure at {currentpress}")
		lastpress = currentpress
	
	print(f"Reached pressure, now at {currentpress}"\n)
	print(datetime.now())

	#g.cset(Pcell = nompress)
    
    def waitfor_move()
        g.waitfor_move()`

**Comment by rerpha** [comment]
I think seeing as a lot of the functionality of the routines procedures rely on the pearl pressure controller being finished we should put this in impeded for now

**Comment by ThomasLohnert** [comment]
Create a new ticket for routines involving pressure controller

**Comment by davidkeymer** [comment]
Ticket was created at the time, [here](https://github.com/ISISComputingGroup/IBEX/issues/6897)

**Comment by daryakoskeroglu** [comment]
https://github.com/ISISComputingGroup/IBEX/pull/7010
